main.py

```diff
@@ -150,13 +154,22 @@ def generate_predictions(methods: list[str], datasets: list[str]) -> Generator[R
 
                 predictions_per_dataset.append(report.prediction)
                 reports_per_dataset.append(report)
+                yield reports_per_dataset
```

Comment:
Ah, now I think it's indented too far. `yield` here will break out of the loop too early.
It needs to be the very last thing executed in the `for dataset_name in datasets:` loop.

Reply:
Ideally:
```
if (np.array(metrics_stat)>-10).all():
                results_store.append(reports_per_dataset)
yield reports_per_dataset
```

Reply:
So at the same indentation as the if statement stub.

Reply:
Done, should be ok now.